host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -1448,7 +1594,30 @@ static MMAL_STATUS_T create_camera_component(RASPIVID_STATE *state)
    // Set the encode format on the video  port
 
    format = video_port->format;
-   format->encoding_variant = MMAL_ENCODING_I420;
+
+   if (state->raw_output)
+   {
+      switch (state->raw_output_fmt)
```

Comment:
Yes, still not fully optimal, but much lower overhead compared to the YUV to RGB conversions you did have.

If you do switch to output[0], then both sets of buffers should be timestamped, so you should be able to tally them. There's no difference at all in the actual image data sent from the two ports - it was more to match the OpenMaxIL spec where the camera component had 2 ports, one for preview and one for capture (extended to 3 ports in a change that I don't believe ever made it into an updated version of the spec).

You'd have an issue even if using output[1], as motion vectors are post video_encode and that has an input FIFO and processing latency. There is unlikely to be any significant difference between that and the processing delay if you use output[0].